klippy/klippy.py

```diff
@@ -57,8 +57,19 @@ def __init__(self, input_fd, bglogger, start_args):
         self.in_shutdown_state = False
         self.run_result = None
         self.event_handlers = {}
-        gc = gcode.GCodeParser(self, input_fd)
-        self.objects = collections.OrderedDict({'gcode': gc})
+        web_hooks = webhooks.WebHooks(self)
+        gc = gcode.GCodeParser(self, input_fd, web_hooks)
+        self.objects = collections.OrderedDict(
+            {'gcode': gc, 'webhooks': web_hooks})
+        # Register Printer Endpoints
+        web_hooks.register_endpoint(
+            "klippy/info", self._handle_info_request)
+        # Endpoint for log file, does not require a callback
+        log_file = start_args.get('log_file')
+        if log_file is not None:
+            web_hooks.register_endpoint(
+                "klippy/klippy.log", None,
+                params={'handler': 'FileRequestHandler', 'path': log_file})
```

Comment:
Can you elaborate on this file endpoint support?  Could mainsail/moonraker obtain the log file name from the info endpoint and just serve the file directly without klippy having to provide a pseudo-endpoint?

Reply:
How will the other static files, gcode upload/download, config file upload/download, etc. work?

I was thinking the request would be something like `http://host/moonraker/get_log_file` (that is, not go through klippy at all), but I don't actually know how that would work, so it might not be a good idea.  A `register_static_file()` does sound like an improvement if klippy must know about these files.

Reply:
Once the static file has been registered the server handles all requests for the file.  The server just needs the registration for configuration so it knows where the file is located on disk and and what the API path should be.